src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -186,11 +198,34 @@ def get_existing_version_info(self, version: str) -> Optional[dict]:
                     generic_sql_error_handler(err=err, role=self.package_role)
                     return None
 
-    def drop_application_before_upgrade(self, policy: PolicyBase, is_interactive: bool):
+    def drop_application_before_upgrade(
+        self, policy: PolicyBase, is_interactive: bool, cascade: bool = False
+    ):
         """
         This method will attempt to drop an application object if a previous upgrade fails.
         """
-        user_prompt = "Do you want the Snowflake CLI to drop the existing application object and recreate it?"
+        if cascade:
+            try:
+                if application_objects := self.get_objects_owned_by_application():
+                    application_objects_str = self._application_objects_to_str(
+                        application_objects
+                    )
+                    cc.message(
+                        f"The following objects are owned by application {self.app_name} and need to dropped:\n{application_objects_str}"
+                    )
+            except ProgrammingError as err:
+                if err.errno != 93079 and ERROR_MESSAGE_093079 not in err.msg:
+                    generic_sql_error_handler(err)
```

Comment:
nit: Could you please add a comment here about why this specific error is handled differently? Just for general readability/maintainability in the future.

Reply:
This is actually prompting me to ask about general error handling. I've noticed a few places where we check both the numeric error code and the contents of the message. That seems a little brittle to me, in case the user-facing message changes. Do we not trust the numeric error code enough to have a local set of constants for each code?
i.e.
```python
if err.errno == APPLICATON_NO_LONGER_AVAILABLE
```
or something like that?

Reply:
I would think that it's sufficient, yes. Maybe someone else from the team has context?

Reply:
I think just error number is sufficient, especially if we properly annotate what the constant means (i.e. `ERROR_NO_APP_NO_LONGER_AVAILABLE` is much better than `ERROR_MESSAGE_093079`).

Reply:
I don't think we should be introspecting error messages because it's possible some user-provided stuff leaks through there.

Reply:
ok sounds good. given that this PR is already approved and there are other places we do this, i'll make it a separate PR to fix them all